src/typescript/frontend/src/components/charts/PrivateChart.tsx

```diff
@@ -145,119 +145,117 @@ export const Chart = async (props: ChartContainerProps) => {
+        // Also, update the `open` price to the previous bar's `close` price if it exists.
+        // NOTE: Since `getBars` is called multiple times, this will result in several
+        // bars having incorrect `open` values. This isn't a big deal but may result in
+        // some visual inconsistencies in the chart.
+        const bars: Bar[] = data.reduce((acc: Bar[], event) => {
+          const bar = toBar(event);
+          if (bar.time >= from * 1000 && bar.time <= to * 1000) {
+            if (acc.at(-1)) {
+              bar.open = acc.at(-1)!.close;
             }
-            bars.push(latestBar);
+            acc.push(bar);
           }
+          return acc;
+        }, []);
 
-          if (bars.length === 0) {
-            if (isFetchForMostRecentBars) {
-              // If this is the most recent bar fetch and there is literally zero trading activity thus far,
-              // we should create a single empty bar to get rid of the `No chart data` ghost error from showing.
-              const time = BigInt(new Date().getTime()) * 1000n;
-              const timeAsPeriod = getPeriodStartTimeFromTime(time, periodDuration) / 1000n;
-              bars.push({
-                time: Number(timeAsPeriod),
-                open: 0,
-                high: 0,
-                low: 0,
-                close: 0,
-                volume: 0,
-              });
-            } else {
-              onHistoryCallback([], {
-                noData: true,
-              });
-              return;
-            }
+        // Push the latest bar to the bars array if it exists and update its `open` value to be the previous bar's
+        // `close` if it's not the first/only bar.
+        if (latestBar) {
+          if (bars.at(-1)) {
+            latestBar.open = bars.at(-1)!.close;
           }
+          bars.push(latestBar);
+        }
 
-          onHistoryCallback(bars, {
-            noData: bars.length === 0,
-          });
-        } catch (e) {
-          if (e instanceof Error) {
-            console.warn("[getBars]: Get error", e);
-            onErrorCallback(e.message);
+        if (bars.length === 0) {
+          if (isFetchForMostRecentBars) {
+            // If this is the most recent bar fetch and there is literally zero trading activity thus far,
+            // we should create a single empty bar to get rid of the `No chart data` ghost error from showing.
+            const time = BigInt(new Date().getTime()) * 1000n;
+            const timeAsPeriod = getPeriodStartTimeFromTime(time, periodDuration) / 1000n;
+            bars.push({
+              time: Number(timeAsPeriod.toString()),
+              open: 0,
+              high: 0,
+              low: 0,
+              close: 0,
+              volume: 0,
+            });
+          } else {
+            onHistoryCallback([], {
+              noData: true,
+            });
+            return;
```

Comment:
```suggestion
```

I think this is pointless actually, right? Ignore me if not, but feel free to change + resolve this comment either way

Reply:
Not sure what this is supposed to do TBH.